source/common/upstream/health_checker_impl.h

```diff
@@ -121,6 +135,7 @@ class HttpHealthCheckerImpl : public HealthCheckerImplBase {
   const std::string host_value_;
   absl::optional<std::string> service_name_;
   Router::HeaderParserPtr request_headers_parser_;
+  const std::unique_ptr<HttpStatusChecker> http_status_checker_;
```

Comment:
nit: This doesn't actually need to be an allocated pointer. You can just make this a const inline member variable and just initialize it as part of the initializer list in the constructor?

Reply:
Pointer removed.